platform/ios/src/MGLMapboxEvents.m

```diff
@@ -489,6 +504,9 @@ - (void)postEvents:(NS_ARRAY_OF(MGLMapboxEventAttributes *) *)events {
             }
             [strongSelf pushDebugEvent:MGLEventTypeLocalDebug withAttributes:@{MGLEventKeyLocalDebugDescription: @"post",
                                                                                     @"debug.eventsCount": @(events.count)}];
+            
+            [[UIApplication sharedApplication] endBackgroundTask:_backgroundTaskIdentifier];
+            _backgroundTaskIdentifier = UIBackgroundTaskInvalid;
```

Comment:
It looks like the background task won’t be invalidated if there was a network error (see [L503](https://github.com/mapbox/mapbox-gl-native/pull/6737/files#diff-6e8e31cd0ffe52aeda9c8b2671bcea19R503)), which would mean the app continues on in the background until it’s killed.